instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb

```diff
@@ -30,19 +30,21 @@ def self.subscribe(
 
         subscriber_object = ::ActiveSupport::Notifications.subscribe(pattern, subscriber)
 
-        ::ActiveSupport::Notifications.notifier.synchronize do
-          subscribers = ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern]
-
-          if subscribers.nil?
-            OpenTelemetry.handle_error(
-              message: 'Unable to move OTEL ActiveSupport Notifications subscriber to the front of the notifications list which may cause incomplete traces.' \
-                       'Please report an issue here: ' \
-                       'https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/new?labels=bug&template=bug_report.md&title=ActiveSupport%20Notifications%20subscribers%20list%20is%20nil'
-            )
-          else
-            subscribers.unshift(
-              subscribers.delete(subscriber_object)
-            )
+        if ::ActiveSupport::Notifications.notifier.respond_to?(:synchronize)
```

Comment:
Do we want to leave any comments to our future selves that we can remove this block once everything prior to 7.2 is EoL?

Reply:
I'm happy to add a comment!